web/src/settings_users.ts

```diff
@@ -0,0 +1,741 @@
+            section.deactivated.list_widget = ListWidget.create(
+                $deactivated_users_table,
+                deactivated_users,
+                {
+                    name: "deactivated_users_table_list",
+                    get_item: people.get_by_user_id,
+                    modifier_html(item) {
+                        const info = human_info(item);
+                        return render_admin_user_list({
+                            ...info,
+                            display_last_active_column: false,
+                        });
+                    },
+                    filter: {
+                        predicate(person) {
+                            return people.predicate_for_user_settings_filters(
+                                person,
+                                section.deactivated.filters,
+                            );
+                        },
+                        onupdate: reset_scrollbar($deactivated_users_table),
+                    },
+                    $parent_container: $("#admin-deactivated-users-list").expectOne(),
+                    init_sort: "full_name_alphabetic",
+                    sort_fields: {
+                        email: user_sort.sort_email,
+                        role: user_sort.sort_role,
+                        id: user_sort.sort_user_id,
+                        ...ListWidget.generic_sort_functions("alphabetic", ["full_name"]),
+                    },
+                    $simplebar_container: $(
+                        "#admin-deactivated-users-list .progressive-table-wrapper",
+                    ),
+                },
+            );
+
+            set_text_search_value(
+                $deactivated_users_table,
+                section.deactivated.filters.text_search,
+            );
+            loading.destroy_indicator($("#admin_page_deactivated_users_loading_indicator"));
+            $("#admin_deactivated_users_table").show();
+        },
+        handle_events() {
+            const $tbody = $("#admin_deactivated_users_table").expectOne();
+
+            handle_filter_change($tbody, section.deactivated);
+            handle_deactivation($tbody);
+            handle_reactivation($tbody);
+            handle_edit_form($tbody);
+        },
+    },
+    bots: {
+        create_table() {
+            loading.make_indicator($("#admin_page_bots_loading_indicator"), {
+                text: $t({defaultMessage: "Loading…"}),
+            });
+            const $bots_table = $("#admin_bots_table");
+            $bots_table.hide();
+            const bot_user_ids = people.get_bot_ids();
+
+            bot_list_widget = ListWidget.create($bots_table, bot_user_ids, {
+                name: "admin_bot_list",
+                get_item: bot_info,
+                modifier_html: render_admin_user_list,
+                html_selector(item) {
+                    const user_id = item?.user_id?.toString() ?? "";
```

Comment:
This looks a little wierd. In what conditions will `item` not contain `user_id` and why should we default this `""`?

Reply:
bot_info can return `undefined`. So `item` can possibly be `undefined` in that case. This might be an ugly way to handle this but I don't think we can use assert in this case.

Reply:
Yeah, I think we can use `assert`